components/modal/frame.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, compose, createRef } from '@wordpress/element';
+import { keycodes } from '@wordpress/utils';
+import { focus } from '@wordpress/dom';
+
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import withFocusReturn from '../higher-order/with-focus-return';
+import withConstrainedTabbing from '../higher-order/with-constrained-tabbing';
+import withGlobalEvents from '../higher-order/with-global-events';
+
+const {
+	ESCAPE,
+} = keycodes;
+
+class ModalFrame extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.containerRef = createRef();
+		this.handleKeyDown = this.handleKeyDown.bind( this );
+		this.handleClickOutside = this.handleClickOutside.bind( this );
+		this.focusFirstTabbable = this.focusFirstTabbable.bind( this );
+	}
+
+	/**
+	 * Focuses the first tabbable element when props.focusOnMount is true.
+	 */
+	componentDidMount() {
+		// Focus on mount
+		if ( this.props.focusOnMount ) {
```

Comment:
The equivalent prop for Popover was enhanced in #7503 to support options. Should we want the same here?

https://github.com/WordPress/gutenberg/blob/master/components/popover/README.md#focusonmount

Reply:
This could be an improvement for a next iteration. I've discussed this with @afercia extensively and the current behaviour is good in terms of a11y.

Reply:
>This could be an improvement for a next iteration.

Please make an issue for this, ideally self-assigned.

Reply:
https://github.com/WordPress/gutenberg/issues/7698